src/core/farcasterTypes.ts

```diff
@@ -0,0 +1,39 @@
+export interface FrameRequest {
+  untrustedData: FrameData;
+  trustedData: {
+    messageBytes: string;
+  };
+}
+
+export interface FrameValidationResponse {
```

Comment:
I think this could be a tagged union, like:
```ts
type FrameValidationResponse =
  | { isValid: true, data: FrameData }
  | { isValid: false, data: undefined }
```

Then if `isValid` is true the typechecker should know that data is defined and vice-versa

Reply:
that is pretty cool